server/utils/cloudFunctions/functions/test/retryAssertions.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Description: retry wrapper for testing assertions that require
+ *              time to pass for an effect to propagate across 
+ *              multiple services. 
+ * Date: 7/22/2020
+ * Author: Rob Marcus  
+ */
+
+const { chai, TIMEOUT_MAX } = require('./testConfig');
+const { promiseTimeout } = require('./promiseTimeout');
+
+/**
+ * Wraps an async function/`=> const` with recursive try and catch logic.
+ * If `timeout` grows larger than `TIMEOUT_MAX`, the test will fail
+ * and the error message will be returned as the reason for test failure.
+ * @param {function} assertions some async (lambda/big arrow) function  
+ * @param {integer} timeout number of ms to wait before retrying the assertions
+ */
+const retryAssertions = async function(assertions, timeout) {
+  try {
+    await assertions();
+  } catch (err) {
```

Comment:
You should check the type of the error and only retry on the algolia error.  Otherwise if your test legitimately fails, it's going to throw an error, and you're going to keep retrying until 30 seconds have been wasted repeatedly retrying the test.

Reply:
I tried to implement this but it is proving difficult - a surprising amount of variation in the valid errors going by during the retry process. Will save this for a future PR.